pkg/model/retry_policy.go

```diff
@@ -0,0 +1,47 @@
+package model
+
+import (
+	"fmt"
+	"time"
+)
+
+// RetryPolicy defines the configuration for retry attempts in case of failures.
+// @Description RetryPolicy defines the configuration for retry attempts in case of failures.
+type RetryPolicy struct {
+	// BaseTimeout is the initial delay between retry attempts, in milliseconds.
+	BaseTimeout int64 `json:"baseTimeout" yaml:"baseTimeout"`
+
+	// Multiplier is used to increase the delay between subsequent retry attempts.
+	// The actual delay is calculated as: BaseTimeout * (Multiplier ^ attemptNumber)
+	Multiplier float64 `json:"multiplier" yaml:"multiplier"`
+
+	// MaxRetries is the maximum number of retry attempts that will be made.
+	// If set to 0, no retries will be performed.
+	MaxRetries int `json:"maxRetries" yaml:"maxRetries"`
+}
+
+// Validate checks if the RetryPolicy fields are valid.
+func (rp *RetryPolicy) Validate() error {
+	if rp == nil {
+		return nil
+	}
+
+	if rp.BaseTimeout <= 0 {
```

Comment:
Why not uint64? if it must be positive

Reply:
This is DTO object.
What if user passes -1? It will be decoded into a really big int.